microcosm_flask/conventions/upload.py

```diff
@@ -0,0 +1,44 @@
+"""
+Conventions for file upload.
+
+"""
+from flask import request
+from microcosm_flask.conventions.base import Convention
+from microcosm_flask.operations import Operation
+from werkzeug.exceptions import BadRequest
+
+
+class UploadConvention(Convention):
+
+    def configure_upload(self, ns, definition):
+        """
+        Register an upload endpoint.
+
+        The definition's func should be an upload function, which must:
+        - accept kwargs for path data
+        - accept a dict of files
+
+        :param ns: the namespace
+        :param definition: the endpoint definition
+
+        """
+        @self.graph.route(ns.collection_path, Operation.Upload, ns)
```

Comment:
I'll file a ticket for that, I am guessing it will be useful for the crespi work, and I've been looking for an opportunity to add something to microcosm.